src/test/common/fdc3.app-channels.ts

```diff
@@ -151,7 +151,11 @@ export function createAppChannelTests(cc: ChannelControl<any,any>, documentation
         const errorMessage = `\r\nSteps to reproduce:\r\n- App A retrieves an app channel\r\n- App B retrieves the same app channel\r\n- App B broadcasts a context of type fdc3.instrument and fdc3.contact\r\n- App A gets current context for types fdc3.instrument and fdc3.contact${documentation}`;
 
         const testChannel = await cc.createTestChannel()
-        await cc.openChannelApp(acTestId9, undefined, APP_CHANNEL_AND_BROADCAST_TWICE)
+        const resolveExecutionCompleteListener = cc.initCompleteListener(acTestId9)
+
+        await cc.openChannelApp(acTestId9, testChannel.id, APP_CHANNEL_AND_BROADCAST_TWICE)
+        await resolveExecutionCompleteListener;
+
         const context = await testChannel.getCurrentContext("fdc3.instrument");
```

Comment:
These test should be extended to also check that adding a context listener does not auto-replay the current context.